src/CSnakes.Tests/GeneratedSignatureTests.cs

```diff
@@ -69,6 +69,8 @@ public void TestGeneratedSignature(string code, string expected)
 
             .AddSyntaxTrees(tree);
         var result = compilation.Emit(testEnv.TempDir + "/HelloWorld.dll");
+        // TODO: Raise compiler warnings as assertion errors
+        result.Diagnostics.Where(d => d.Severity == DiagnosticSeverity.Error).ToList().ForEach(d => Assert.Fail(d.ToString()));

```

Comment:
It'd be better to test that there are no items that match that LINQ statement, test that the collection is empty